src/Runner.Worker/Handlers/StallManager.cs

```diff
@@ -0,0 +1,70 @@
+using System;
+using System.Collections.Generic;
+using System.Timers;
+using GitHub.Runner.Common;
+using GitHub.Runner.Sdk;
+using GitHub.Runner.Worker;
+
+namespace GitHub.Runner.Worker.Handlers
+{
+    [ServiceLocator(Default = typeof(TimerAdapter))]
+    public interface ITimer 
+    {
+        void Start();
+        void Stop();
+        double Interval { get; set; }
+        event ElapsedEventHandler Elapsed;
+        bool AutoReset { get; set; }
+        void Dispose();
+    }
+
+    public class TimerAdapter : Timer, ITimer {}
+
+    public sealed class StallManager: IDisposable
```

Comment:
Should we push all this logic into the processInvoker at the bottom layer, and any caller just needs to say `processInvoker.ExecuateAsync(x,x,x,x, stallDetect: true)`

Reply:
I think this is valuable to do, but would like to do it in another PR as this has been stale for awhile and it's a bunch of reconfiguring